### DISCUSSION TITLE
Why does `subject.pipe(op1(), ...)` returns an `AnonymousSujbect`

### DISCUSSION THREAD

**Discussion by LcsGa** [ReactiveX/rxjs#7337] [Q&A] [answered]
I've always thought that calling the `.pipe()` method to a subject (with at least one operator) resulted in an observable and thus, I didn't need to call `.asObservable` too. It appears that I was wrong as I tried to log something like `subject.pipe(op1())` and it logged an `AnonymousSubject` on which I can still use the `next`, `error` and `complete` methods.

I tried to find out in the source code why this happens but couldn't find anywhere it should transform a subject into an anonymousSubject... Can you please explain to me where this happens :sweat_smile:?

(NB: Here is an example of what I did to figure out this https://stackblitz.com/edit/rxjs-nw7h6x?file=index.ts)

**Comment by voliva** [comment, marked as answer]
This is because `pipe` applies the operators one by one, and those use "lift", which keeps the original constructor intact #60

The only problem is that this can't be easily done with TS, so the type definition for `pipe` returns an Observable, but it actually returns whatever the original observable type's.

Related #6148 and #7201, quoting Ben Lesh:

> The only issue I see is currently RxJS supports the composition of our subject variants through the operator chain, via lift. This is a little known aspect of RxJS that isn't used much, in no smart part because TypeScript can't support it:
>
> ```ts
> const subject = new Subject<number>()
>
> const lifted = subject.pipe(map(n => n + '?'), map(s => s + '!');
>
> lifted.subscribe(console.log);
>
> (lifted as any).next(42); // logs "42?!"
> ```


This won't apply for v8.x, as lift was removed: #7202

**Reply by LcsGa** [reply]
Thank you very much for your very clear answer! 😁
